doc/source/whatsnew/v0.20.3.txt

```diff
@@ -79,7 +79,7 @@ Sparse
 
 Reshaping
 ^^^^^^^^^
-
+- Fixed missing sort for index ``join`` method (:issue: `16541`) 
```

Comment:
``PeriodIndex / TimedeltaIndex.join`` was missing the ``sort=`` kwarg (you can add a ref to the original issue where it was added as well)